crates/router/src/core/payments/operations/payment_capture.rs

```diff
@@ -251,20 +251,29 @@ impl<F: Clone, Ctx: PaymentMethodRetrieve>
     where
         F: 'b + Send,
     {
-        payment_data.payment_attempt = match &payment_data.multiple_capture_data {
-            Some(multiple_capture_data) => db
-                .store
+        payment_data.payment_attempt = if payment_data.multiple_capture_data.is_some()
+            || payment_data.payment_attempt.amount_to_capture.is_some()
+        {
```

Comment:
This can be achieved using match statement
```suggestion
        match &payment_data.multiple_capture_data {
            Some(multiple_capture_data) if payment_data.payment_attempt.amount_to_capture.is_some()
        { }
```

I think this is what you want to achieve right?

Reply:
I wanted to eliminate unnecessary DB Update call. Update DB only if Some data is present in either in `payment_data.multiple_capture_data`, or `payment_data.payment_attempt.amount_to_capture`